node/network/protocol/src/grid_topology.rs

```diff
@@ -142,6 +164,45 @@ impl SessionGridTopologies {
 		}
 	}
 }
+
+/// A simple storage for a topology and the corresponding session index
+#[derive(Default, Debug)]
+pub struct GridTopologySessionBound(SessionGridTopology, SessionIndex);
+
+/// A storage for the current and maybe previous topology
+#[derive(Default, Debug)]
+pub struct SessionBoundGridTopologyStorage {
+	current_topology: GridTopologySessionBound,
+	prev_topology: Option<GridTopologySessionBound>,
+}
+
+impl SessionBoundGridTopologyStorage {
+	/// Return a grid topology based on the session index:
+	/// If we need a previous session and it is registered in the storage, then return that session.
+	/// Otherwise, return a current session to have some grid topology in any case
```

Comment:
We could improve readability by returning an option here and letting the caller do `unwrap_or()` to fetch the current topology when we can't find one for the specified index.  Or we can just rename to smth like `get_topology_or_current`.